src/main/scala/MtUploader.scala

```diff
@@ -10,8 +10,7 @@ import scala.concurrent.{Await, ExecutionContext, Future}
 import scala.util.{Failure, Success, Try}
 import scala.concurrent.duration._
 
-class MtUploader (bucketName: String, removePathSegments: Int){
-  val CHUNK_SIZE:Long = 8*1024*1024
+class MtUploader (bucketName: String, removePathSegments: Int, chunkSize:Long = 8*1024*1024){
```

Comment:
What is the point of this multiplication? Why not just supply 8388608?

Reply:
just because it makes it more explicit that the default chunk size is 8 megs, rather than a long number that at first glance could be 8.3 megs or 83 megs. I would expect the compiler to optimize this to a single static value